lib/connect/api/productActions.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * This file is part of the Ingram Micro Cloud Blue Connect SDK.
+ *
+ * @copyright (c) 2019. Ingram Micro. All Rights Reserved.
+ */
+
+/* eslint-disable max-len */
+
+const BaseService = require('./base');
+const { addFilterParam } = require('./utils');
+/**
+* The ActionService class provides methods to access the ``Action``
+* endpoint of the Cloud Blue Connect API.
+*/
+class ActionService extends BaseService {
+  /**
+   * Get a list of actions related to a product and optionally version of the product.
+   *
+   * @param  {string} productId                                    The unique identifier of the product these actions belongs to.
+   * @param  {string} productVersion                               The optional product's version identifier.
+   * @param  {number} [limit=100]                                  The maximum number of records to return.
+   * @param  {number} [offset=0]                                   The offset from which to start to include records.
+   * @return {Array} An array of Action objects.
+   */
+  async list(productId, productVersion, limit = 100, offset = 0) { /* eslint-enable max-len */
+    const params = { limit, offset };
+
+    let url = `/products/${productId}`;
+    if (productVersion) {
+      url = `${url}/versions/${productVersion}`;
+    }
+
+    url = `${url}/actions`;
+    url = this.addParams(url, params);
```

Comment:
This looks quite tricky and "procedure" style for me
Let us refactor this to a separate function so we could have something like this
`const url = getProductUrl(productVersion, params)`
Let us move all related branches and transformations inside. Will be more declarative.

Reply:
Same for other places here
I'm pretty sure we even can make something reusable for all related places